### PR TITLE
Add route to v1beta3

### PR DESCRIPTION
**Pull request by smarterclayton**
@pweil-

### CODE REVIEW

File: pkg/route/api/v1beta3/types.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+)
+
+// Route encapsulates the inputs needed to connect an alias to endpoints.
+type Route struct {
+	kapi.TypeMeta   `json:",inline"`
+	kapi.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RouteSpec   `json:"spec"`
```
Comment (pweil-) [inline pkg/route/api/v1beta3/types.go]: This just made my day :tada: :balloon: :cake:

File: pkg/route/api/v1beta3/types.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	TargetName string `json:"targetName"`
+
+	// Optional: the value of the target endpoint port to expose. May be omitted if
+	// name is set, and vice versa.
+	TargetPort util.IntOrString `json:"targetPort"`
+}
+*/
+
+// RouteStatus describes the current state of this route.
+type RouteStatus struct{}
```
Comment (pweil-) [inline pkg/route/api/v1beta3/types.go]: We can probably go ahead and add a reference to the RouterShard here unless the plan is to hold it in an annotation.

Statuses that I can think of: Pending, Allocated, Accepted, Running/Deployed/Live, Failed

Accepted would be a short transition state where the router has received the route but has not yet written config/recycled the router.

**Comment by pweil-** [comment]
This looks good.  I take it we're keeping the internal api at v1beta1 until everything else is added?  That threw me for a loop initially.

**Comment by smarterclayton** [comment]
Yeah, this is pure v1beta1 reorganized.  We can do the internal refactor later (probably at the same time we kill the old objects).  That's a lot less expensive than doing the internal refactor first.

----- Original Message -----

File: pkg/route/api/v1beta3/types.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/route/api/v1beta3/types.go]: Until we know how the shard is going to be organized (will we do Spec.Shard like we do for Pods, or Spec.Shards[]) I want to hold off.

----- Original Message -----

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**